framework/db/QueryTrait.php

```diff
@@ -363,14 +363,21 @@ public function addOrderBy($columns)
      */
     protected function normalizeOrderBy($columns)
     {
+        $result = [];
         if ($columns instanceof ExpressionInterface) {
             return [$columns];
         } elseif (is_array($columns)) {
-            return $columns;
+            foreach ($columns as $column => $value) {
+                if (preg_match('#(asc|desc)#i', $value, $matches)) {
```

Comment:
How about ORDER BY description or ORDER BY ascription?

Reply:
It could search for `" asc"` and `" desc"`. In this case non-regex search could be a bit faster. Or it could use word boundaries i.e. `#\b(asc|desc)\b#i`.

Reply:
Oh, your right. Maybe add  `^...$` ?
`#^\b(asc|desc)\b$#i`

Reply:
Without ^ $ like a trim. I will add your solution. Thanks!